### PR TITLE
Add GitHub Actions integration workflow

### LINKED ISSUE #777: Switch from Travis CI to GitHub Actions
~At the moment, this repo is using Travis CI, which runs rather slow. This issue aims to switch from Travis CI to Circle CI.~
~Example config file: https://github.com/Automattic/vip-go-mu-plugins/blob/master/.circleci/config.yml~

Update: 

At the moment, this repo is using Travis CI, which runs rather slow. This issue aims to switch from Travis CI to GitHub Actions.

Scope:

- [ ] Run plugin against single- and multisite
- [x] Run plugin against PHPCS (PHP 5.6 - PHP 8.0)
- [x] Run plugin against PHPUnit (PHP 7.1 - PHP 7.4)

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #777 

**Changes**

Add GitHub Actions integration workflow to switch from Travis CI to GitHub Actions.

**Steps to test**

1. Head over to https://github.com/Automattic/Co-Authors-Plus/actions
2. Look up the run(s)

**Note**

- ~PHPUnit only runs against PHP 7.1 till PHP 7.4~
- ~PHPUnit shows two failed unit tests which will be solved with #774~
- ~PHPCS runs against PHP 5.6 till PHP 8.0~
- PHPCS shows various errors which should be addressed in a separate issue
- The `WP 5.7 on PHP 8.1` run is allowed to fail, as [PHP 8.1 hasn't been released yet](https://wiki.php.net/todo/php81)
- The `WP 5.5 on PHP 8.0` run had been excluded as [PHP 8.0 is supported for WP > 5.5](https://make.wordpress.org/core/2020/11/23/wordpress-and-php-8-0/)

### CODE REVIEW

**Review by rebeccahum — COMMENTED** [review]
Do we still need the .travis.yml file? https://github.com/Automattic/Co-Authors-Plus/blob/master/.travis.yml

**Comment by nielslange** [comment]
> Do we still need the .travis.yml file? https://github.com/Automattic/Co-Authors-Plus/blob/master/.travis.yml

Not really. I'll delete the `.travis.yml` file as part of this PR.

**Comment by nielslange** [comment]
@rebeccahum While looking at the final implementation, I noticed that `phpcs` was not running in the `.travis.yml` file. Running the change against `phpcs` leads to various errors. I have three options in mind to address this case:

1. Exclude `phpcs` from the current PR as `phpcs` is not running in the `.travis.yml` file either. This leads to a run in which all required tests are passing (see https://github.com/Automattic/Co-Authors-Plus/actions/runs/776485188).
2. Include `phpcs` in the current PR to ensure that the plugin aligns with the VIP Go coding standards. This leads to a run in which none of the tests are passing (see https://github.com/Automattic/Co-Authors-Plus/actions/runs/776570819).
3. Split the file `integration.yml` into `phpunit.yml` and `phpcs.yml` to run these two scenarios independent of each other. While this leads to duplicate code, it also shows clearly which tests are passing and which not (see https://github.com/Automattic/Co-Authors-Plus/actions/runs/776611607 and https://github.com/Automattic/Co-Authors-Plus/actions/runs/776611609).

What are your thoughts on that?

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          php --version
+          ./vendor/bin/phpunit --version
+
+      - name: Start MySQL service
+        run: sudo /etc/init.d/mysql start
+
+      - name: Use mysql_native_password for PHP < 7.4
+        if: ${{ matrix.php < 7.4 }}
+        run: mysql -u root -p"root" -e "ALTER USER 'root'@localhost IDENTIFIED WITH mysql_native_password BY 'root'";
+
```
Comment (htdat) [inline .github/workflows/phpunit.yml:66]: @nielslange - this run is not necessary anymore as you're using `ubuntu-16.04`

Comment (nielslange) [reply .github/workflows/phpunit.yml:66]: Cool. Thanks for clarifying this, @htdat.

File: tests/test-coauthors-guest-authors.php [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 class Test_CoAuthors_Guest_Authors extends CoAuthorsPlus_TestCase {
 
+	use Yoast\PHPUnitPolyfills\Polyfills\AssertStringContains;
```
Comment (rebeccahum) [inline tests/test-coauthors-guest-authors.php:5]: This seems irrelevant to the current PR. Did it accidentally sneak in?

File: tests/test-template-tags.php [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 class Test_Template_Tags extends CoAuthorsPlus_TestCase {
 
+	use Yoast\PHPUnitPolyfills\Polyfills\AssertStringContains;
```
Comment (rebeccahum) [inline tests/test-template-tags.php:5]: Same as above ^^.

File: composer.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-	"name"       : "automattic/co-authors-plus",
+    "name"       : "automattic/co-authors-plus",
```
Comment (rebeccahum) [inline composer.json:2]: Formatting looks off here.

**Review by rebeccahum — CHANGES_REQUESTED** [review]

**Comment by rebeccahum** [comment]
@nielslange I'm not sure I'm understanding the benefits towards splitting it if the tests will still fail. 

> Split the file integration.yml into phpunit.yml and phpcs.yml to run these two scenarios independent of each other. While this leads to duplicate code, it also shows clearly which tests are passing and which not (see https://github.com/Automattic/Co-Authors-Plus/actions/runs/776611607 and https://github.com/Automattic/Co-Authors-Plus/actions/runs/776611609).

It might be better to adjust CAP so that tests pass - we can do that by modifying the code towards standards or if not applicable, adding PHPCS ignore statements.

File: tests/test-coauthors-guest-authors.php (hunk 2 above)
Comment (nielslange) [reply tests/test-coauthors-guest-authors.php:5]: I added `use Yoast\PHPUnitPolyfills\Polyfills\AssertStringContains;` on purpose. In https://github.com/Automattic/Co-Authors-Plus/pull/774#discussion_r610490537, @GaryJones pointed out using `assertStringContainsString()` over `assertContains()`. `assertContains()` was soft deprecated with PHPUnit 7.5.0, hard depricated with PHPUnit 8.0.0 and removed with PHPUnit 9.0.0. On the other hand, `assertStringContainsString()` had been introduced with PHPUnit 7.5.0. As I'm also running tests against PHP 5.6 with PHPUnit 5.7.27, I need [PHPUnit-Polyfills](https://github.com/Yoast/PHPUnit-Polyfills) to be able to run the unit tests with various PHPUnit versions.

File: tests/test-template-tags.php (hunk 3 above)
Comment (nielslange) [reply tests/test-template-tags.php:5]: Same as above.

File: composer.json (hunk 4 above)
Comment (nielslange) [reply composer.json]: I noticed that the file was not correctly formatted in general and just adjusted this.

Comment (GaryJones) [reply composer.json]: Use a `.editorconfig` file, and https://github.com/ergebnis/composer-normalize

File: .github/workflows/phpcs.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [ '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+        wordpress: [ '5.5', '5.6', '5.7'  ]
+        allowed_failure: [ false ]
+        include:
+          # PHP nightly.
+          - php: '8.1'
```
Comment (rebeccahum) [inline .github/workflows/phpcs.yml:22]: It might be good to exclude the nightly for now as our dependencies aren't compatible with it else we just risk seeing consistent failures (even though they are allowed).

**Review by rebeccahum — CHANGES_REQUESTED** [review]

File: .github/workflows/phpunit.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [ '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+        wordpress: [ '5.5', '5.6', '5.7'  ]
+        allowed_failure: [ false ]
+        include:
+          # PHP nightly.
+          - php: '8.1'
```
Comment (rebeccahum) [inline .github/workflows/phpunit.yml:22]: Same as above ^^.

File: .github/workflows/phpcs.yml (hunk 5 above)
Comment (nielslange) [reply .github/workflows/phpcs.yml]: Sounds good. I just removed the nightly PHP version.

File: .github/workflows/phpunit.yml (hunk 6 above)
Comment (nielslange) [reply .github/workflows/phpunit.yml]: See https://github.com/Automattic/Co-Authors-Plus/pull/784#discussion_r625997673

**Review by rebeccahum — APPROVED** [review]
